sdk/java/dagger-java-annotation-processor/src/test/resources/io/dagger/java/module/DaggerJava.java

```diff
@@ -14,6 +14,12 @@
 /** Dagger Java Module main object */
 @Object
 public class DaggerJava extends AbstractModule {
+  /** Project source directory */
+  private Directory source;
+
+  /** Go version */
+  private String version;
```

Comment:
I think if it's private it shouldn't be exposed to Dagger.

See https://docs.dagger.io/api/state

If you compare with Go, notice that sometimes `// +private` is used. That's because of the serialization/deserialization, which only works on public fields of a struct so that pragma is needed when a user needs the serialization but not to expose it to Dagger. Otherwise, just making it private (i.e., camelCase instead of PascalCase) is enough to not expose it.

Reply:
I'm talking about the one the SDK client does when getting `parentJson` and returning object results.